packages/byods/src/token-storage-adapter/index.ts

```diff
@@ -0,0 +1,77 @@
+import {OrgServiceAppAuthorization} from '../types';
+import {TokenStorageAdapter} from './types';
+
+/**
+ * An in-memory adapter for storing, listing and retrieving service app authorization tokens.
+ *
+ * @public
+ */
+export class InMemoryTokenStorageAdapter implements TokenStorageAdapter {
+  /**
+   * The local memory cache for the tokens.
+   */
+  private tokenCache: {[orgId: string]: OrgServiceAppAuthorization};
+
+  constructor(tokenCache: {[orgId: string]: OrgServiceAppAuthorization} = {}) {
+    this.tokenCache = tokenCache;
+  }
+
+  /**
+   * Method to set the token for an organization.
+   * @param orgId Organization ID
+   * @param token Respective token
+   * @example
+   * await storageAdapter.setToken('org-id', token);
+   */
+  async setToken(orgId: string, token: OrgServiceAppAuthorization): Promise<void> {
+    this.tokenCache[orgId] = token;
+  }
+
+  /**
+   * Method to extract a token based on the organization id.
+   * @param orgId Organization ID
+   * @returns {OrgServiceAppAuthorization} The token object for the organization
+   * @example
+   * const token = await storageAdapter.getToken('org-id');
+   */
+  async getToken(orgId: string): Promise<OrgServiceAppAuthorization> {
+    if (this.tokenCache[orgId]) {
+      return this.tokenCache[orgId];
+    }
+    throw new Error(`Service App token not found for org ID: ${orgId}`);
+  }
+
+  /**
+   * Method which returns the list of all tokens stored in the InMemoryTokenStorageAdapter.
+   * @returns {OrgServiceAppAuthorization[]} List of
+   * @example
+   * const tokens = await storageAdapter.listTokens();
+   */
+  async listTokens(): Promise<OrgServiceAppAuthorization[]> {
+    return Object.values(this.tokenCache);
```

Comment:
Same here, if it is empty, it will return an empty list - no need to throw an error for this.
Open to suggestions, however.